File: src/com/suushiemaniac/lang/json/util/StringUtils.java
package com.suushiemaniac.lang.json.util;

public class StringUtils {
    public static int compare(String s1, String s2) {
        return s1.compareTo(s2);
    }

    public static String copy(String base, int times) {
        String cp = "";

        for (int i = 0; i < times; i++) cp += base;

        return cp;
    }

    public static String jsonWrap(String blank) {
        if (!blank.startsWith("\"") && !blank.endsWith("\"")) blank = "\"" + blank + "\"";
        return blank;
    }

    public static String jsonUnwrap(String wrapped) {
        if (wrapped.startsWith("\"") && wrapped.endsWith("\"")) wrapped = wrapped.substring(1, wrapped.length() - 1);
        return wrapped;
    }

    public static String reduceTab(String tabbed) {
        while (tabbed.startsWith("\t")) tabbed = tabbed.substring(1);
        return tabbed;
    }
}

File: src/main/kotlin/com/suushiemaniac/lang/json/util/StringUtils.kt
package com.suushiemaniac.lang.json.util

object StringUtils {
    private val UNICODE = "\\\\u([0-9a-fA-F]{4})".toRegex()

    fun String.jsonWrap(): String {
        var blank = this.replace("\\", "\\\\").replace("\"", "\\\"")

        if (!blank.startsWith("\"") || !blank.endsWith("\"")) {
            blank = "\"" + blank + "\""
        }

        return blank
    }

    fun String.jsonUnwrap(): String {
        if (this.startsWith("\"") && this.endsWith("\"") && this.length > 1) {
            return this.substring(1, this.length - 1)
        }

        return this
    }

    fun String.reduceTab(): String {
        var tabbed = String(this.toCharArray())

        while (tabbed.startsWith("\t")) {
            tabbed = tabbed.substring(1)
        }

        return tabbed
    }

    fun String.decodeUnicode(): String {
        var decoded = String(this.toCharArray())
        val unicodeMatcher = UNICODE.findAll(decoded)

        for (res in unicodeMatcher) {
            val num = Integer.parseInt(res.groupValues[1], 16)
            decoded = decoded.replace(Regex.escape(res.value).toRegex(), num.toChar().toString())
        }

        return decoded
    }

    fun String.encodeUnicode(): String {
        val b = StringBuilder()

        for (c in this.toCharArray()) {
            if (c.toInt() >= 256) {
                b.append("\\u").append(String.format("%04X", c.toInt()))
            } else {
                b.append(c)
            }
        }

        return b.toString()
    }
}
